src/ServiceManagerBase.sol

```diff
@@ -112,14 +107,62 @@ abstract contract ServiceManagerBase is OwnableUpgradeable, ServiceManagerBaseSt
                 strategies[j] = strategy;
             }
 
-            _operatorSetManager.addStrategiesToOperatorSet(
-                uint32(i),
-                strategies  
-            );
+            _operatorSetManager.addStrategiesToOperatorSet(uint32(i), strategies);
+            emit OperatorSetStrategiesMigrated(uint32(i), strategies);
         }
         isStrategiesMigrated = true;
     }
 
+    /**
+     * @notice the operator needs to provide a signature over the operatorSetIds they will be registering
+     * for. This can be called externally by getOperatorSetIds
+     */
+    function migrateOperatorToOperatorSets(
+        address operator,
+        ISignatureUtils.SignatureWithSaltAndExpiry memory signature
+    ) external {
+        require(
+            !isOperatorMigrated[operator],
+            "ServiceManagerBase.migrateOperatorToOperatorSets: already migrated"
+        );
+        uint32[] memory operatorSetIds = getOperatorSetIds(operator);
+
+        _operatorSetManager.registerOperatorToOperatorSets(operator, operatorSetIds, signature);
+        emit OperatorMigratedToOperatorSets(operator, operatorSetIds);
+    }
+
+    /**
+     * @notice Once strategies have been migrated and operators have been migrated to operator sets.
+     * The ServiceManager owner can eject any operators that have yet to completely migrate fully to operator sets.
+     * This final step of the migration process will ensure the full migration of all operators to operator sets.
+     * @param operators The list of operators to eject for the given OperatorSet
+     * @param operatorSet The OperatorSet to eject the operators from
+     * @dev The RegistryCoordinator MUST set this ServiceManager contract to be the ejector address for this call to succeed
+     */
+    function ejectNonmigratedOperators(
+        address[] calldata operators,
+        IOperatorSetManager.OperatorSet calldata operatorSet
+    ) external onlyOwner {
+        require(
+            operatorSet.avs == address(this),
+            "ServiceManagerBase.ejectNonmigratedOperators: operatorSet is not for this AVS"
+        );
+        require(
+            operatorSet.id < _registryCoordinator.quorumCount(),
+            "ServiceManagerBase.ejectNonmigratedOperators: operatorSet does not exist"
+        );
+        for (uint256 i = 0; i < operators.length; ++i) {
+            require(
+                !_operatorSetManager.isOperatorInOperatorSet(operators[i], operatorSet),
```

Comment:
we need to check that they're actually registered in the AVS for the given operator set

Reply:
Registered in the Registry coordinator? Or in the OSM? Because ejectOperator will handle the former

Reply:
https://github.com/Layr-Labs/eigenlayer-middleware/pull/263/commits/78f35d4b5743c8532a226d47ed2ce90ea38fbfa1

Checking their AVS registration status is REGISTERED and that they are not in the operatorSet